src/components/message-input/emoji-picker.tsx

```diff
@@ -0,0 +1,76 @@
+import React, { RefObject } from 'react';
+import { mentionsConfigs } from './mentions-config';
+import { Picker } from 'emoji-mart';
+import { ViewModes } from '../../shared-components/theme-engine';
+import { mapPlainTextIndex } from './react-mentions-utils';
+
+import 'emoji-mart/css/emoji-mart.css';
+
+export interface Properties {
+  textareaRef: RefObject<HTMLTextAreaElement>;
+  isOpen: boolean;
+  onOpen: () => void;
+  onClose: () => void;
+  value: string;
+  viewMode: ViewModes;
+  onSelect: (value: string) => void;
+}
+
+export class EmojiPicker extends React.Component<Properties> {
+  componentDidMount() {
+    document.addEventListener('mousedown', this.clickOutsideEmojiCheck);
+  }
+
+  componentWillUnmount() {
+    document.removeEventListener('mousedown', this.clickOutsideEmojiCheck);
+  }
+
+  clickOutsideEmojiCheck = (event: MouseEvent) => {
+    if (!this.props.isOpen) {
+      return;
+    }
+
+    const [emojiMart] = document.getElementsByClassName('emoji-mart');
+
+    if (emojiMart && event && event.target) {
+      if (!emojiMart.contains(event.target as Node)) {
+        this.props.onClose();
+      }
+    }
+  };
+
+  insertEmoji = (emoji: any) => {
+    const emojiToInsert = emoji.colons + ' ';
```

Comment:
We should use the native character here and not the token.

Reply:
https://github.com/m3m3n70/zero-web/blob/development/src/components/message-input/index.tsx#L255